File: c276(break_time)/Python/break_time.py
while True:
    try:
        password = input()
        try_times = int(input())
        match = []
        for i in range(try_times):
            try_password = input()
            try_count = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
            count = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
            a, b = 0, 0
            for j in range(4):
                if password[j] == try_password[j]:
                    a += 1
                else:
                    try_count[int(try_password[j]) % 10] += 1
                    count[int(password[j]) % 10] += 1
            for j in range(10):
                if try_count[j] != 0 and count[j]:
                    b += (try_count[j] if try_count[j] < count[j] else count[j])
                    if b >= (4 - a): break
            match.append(f'{a}A{b}B')
        for try_case in match:
            print(try_case)
    except:
        break

File: c276(break_time)/C++/break_time.cpp
#include <iostream>

int main(int argc, char const *argv[]) {
  std::ios_base::sync_with_stdio(false);
  std::cin.tie(0);
  std::string password;
  while (std::cin >> password) {
    int try_times;
    std::string try_password;
    std::cin >> try_times;
    while (try_times != 0) {
      int count[10] = {0}, try_count[10] = {0};
      int a = 0, b = 0;
      --try_times;
      std::cin >> try_password;
      for (int i = 0; i < 4; i++) {
        if (try_password[i] == password[i]) ++a;
        else {
          ++try_count[((int)try_password[i] - 48) % 10];
          ++count[((int)password[i] - 48) % 10];
        }
      }
      for (int i = 0; i < 10; i++) {
        if (try_count[i] != 0 && count[i] != 0) {
          b += (try_count[i] > count[i] ? count[i] : try_count[i]);
          if (b >= 4 - a) break;
        }
      }
      std::cout << a << 'A' << b << "B\n";
    }
    std::cout << std::flush; // optional
  }

  return 0;
}
